lib/models.js

```diff
@@ -1141,6 +1141,47 @@ class TypeConstraint extends Constraint {
   }
 }
 
+class SubsetConstraint extends Constraint {
+  constructor(subsetList, path, onValue=false) {
+    super(path);
+    this._subsetList = subsetList;
+    this._onValue = onValue;
+  }
+
+  get onValue() { return this._onValue; }
+  set onValue(onValue) {
+    this._onValue = onValue;
+  }
+
+  // withOnValue is a convenience function for chaining
+  withOnValue(onValue) {
+    this.onValue = onValue;
+    return this;
+  }
+
+  get subsetList() { return this._subsetList; }
+
+  clone() {
+    const newList = this._subsetList.map(c => c.clone());
+    const clone = new SubsetConstraint(newList);
+    clone.onValue = this.onValue;
+    this._clonePropertiesTo(clone);
+    return clone;
+  }
+
+  equals(other) {
+    return (other instanceof SubsetConstraint) &&
+        this._onValue === other.onValue &&
+        this._pathsAreEqual(other) &&
+        this._subsetList.length === other.subsetList.length &&
+        this._subsetList.every((s, i) => s === other.subsetList[i]);
```

Comment:
If `subsetList` contains `Identifier`s, then we can't compare with `===`.  We should be able to use the `.equals(other)` function on `Identifier` though.

Reply:
Made this change, and updated `getPossibleIdentifiers` here https://github.com/standardhealth/shr-models/pull/47/commits/11e716ce21a2030f2b55e48d5a74c330712b7afa.